packages/http-signature-utils/src/utils/verification.ts

```diff
@@ -0,0 +1,136 @@
+import * as crypto from 'crypto'
+import { verifyContentDigest } from 'httpbis-digest-headers'
+import { importJWK } from 'jose'
+import { Context } from 'koa'
+import { JWK } from './jwk'
+
+type HttpSigHeaders = Record<'signature' | 'signature-input', string>
+
+type HttpSigRequest = Omit<Context['request'], 'headers'> & {
+  headers: HttpSigHeaders
+}
+
+export type HttpSigContext = Context & {
+  request: HttpSigRequest
+  headers: HttpSigHeaders
+}
+
+export function validateHttpSigHeaders(ctx: Context): ctx is HttpSigContext {
```

Comment:
I'm leaning towards this package's exported methods not using koa contexts.
An alternative could look something like `createSignatureHeaders`'s interface.

Reply:
Good point